sympy/polys/rootoftools.py

```diff
@@ -970,6 +970,10 @@ def _eval_Eq(self, other):
         # is_real value of the CRootOf instance.
         if type(self) == type(other):
             return sympify(self == other)
+        if other.is_number and other.has(AppliedUndef) or not other.is_number:
+            # it's a number that can't be evaluated or it's not a number in its current form
+            # though simplification might identify it as such (but that is the user's choice)
+            return None
         if not (other.is_number and not other.has(AppliedUndef)):
```

Comment:
These are mutually exclusive so if it is a number it doesn't have AppliedUndef so this simplifies to `if not other.is_number`...but that is what the previous if-block is testing so this is redundant.

It *might* make more sense to do
```python
if not other.is_number:
    # it might be a number after simplification but it isn't right now
    # but if it is known that it is not constant, then it can't be a
    # RootOf value (hence False)
    return other.is_constant
```
But I would tend to just return the None and let the user decide if they want to attempt simplification; I know `equals` is expensive, but I forget how expensive `is_constant` is.

Reply:
I didn't change this online because you might want to test it locally before making this change.

Reply:
I'm not sure I understand this. Surely we don't want to return True just because `other.is_constant` is True? That doesn't imply that these two are equal.

Reply:
right: `return None if other.is_constant else False`